lattice-controller/src/main.rs

```diff
@@ -51,290 +54,267 @@ impl Default for ConnectionConfig {
     }
 }
 
-impl ConnectionConfig {
-    fn new_from(
-        values: &HashMap<String, String>,
-        mut nats_uris: Vec<String>,
-    ) -> RpcResult<ConnectionConfig> {
-        let mut config = if let Some(config_b64) = values.get("config_b64") {
-            let bytes = base64::decode(config_b64.as_bytes()).map_err(|e| {
-                RpcError::InvalidParameter(format!("invalid base64 encoding: {}", e))
-            })?;
-            serde_json::from_slice::<ConnectionConfig>(&bytes)
-                .map_err(|e| RpcError::InvalidParameter(format!("corrupt config_b64: {}", e)))?
-        } else if let Some(config) = values.get("config_json") {
-            serde_json::from_str::<ConnectionConfig>(config)
-                .map_err(|e| RpcError::InvalidParameter(format!("corrupt config_json: {}", e)))?
-        } else {
-            ConnectionConfig::default()
-        };
-
-        if let Some(url) = values.get(ENV_NATS_URI) {
-            config.cluster_uris.push(url.clone());
-        }
-        if let Some(jwt) = values.get(ENV_NATS_CLIENT_JWT) {
-            config.auth_jwt = Some(jwt.clone());
-        }
-        if let Some(seed) = values.get(ENV_NATS_CLIENT_SEED) {
-            config.auth_seed = Some(seed.clone());
-        }
-        if config.auth_jwt.is_some() && config.auth_seed.is_none() {
-            return Err(RpcError::InvalidParameter(
-                "if you specify jwt, you must also specify a seed".to_string(),
-            ));
-        }
-        if config.cluster_uris.is_empty() {
-            config.cluster_uris.append(&mut nats_uris);
-        }
-
-        if let Some(nsprefix) = values.get(ENV_LATTICE_PREFIX) {
-            config.lattice_prefix = nsprefix.to_owned();
-        }
-
-        if let Some(auction_timeout) = values.get(ENV_AUCTION_TIMEOUT_MS) {
-            config.auction_timeout_ms = auction_timeout.parse().unwrap_or(3 * DEFAULT_TIMEOUT_MS)
-        }
-
-        if let Some(timeout) = values.get(ENV_TIMEOUT_MS) {
-            config.timeout_ms = timeout.parse().unwrap_or(DEFAULT_TIMEOUT_MS);
-        }
-
-        if config.timeout_ms == 0 {
-            config.timeout_ms = DEFAULT_TIMEOUT_MS
-        }
-
-        if config.auction_timeout_ms == 0 {
-            config.auction_timeout_ms = 3 * DEFAULT_TIMEOUT_MS
-        }
-
-        if config.lattice_prefix.is_empty() {
-            config.lattice_prefix = "default".to_owned()
-        }
-
-        Ok(config)
-    }
-}
-
 // main (via provider_main) initializes the threaded tokio executor,
 // listens to lattice rpcs, handles actor links,
 // and returns only when it receives a shutdown message
-//
-fn main() -> Result<(), Box<dyn std::error::Error>> {
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
```

Comment:
Because of this change, you must change `provider_start` to `provider_run` to avoid initializing another tokio runtime https://docs.rs/wasmbus-rpc/0.9.3/src/wasmbus_rpc/provider_main.rs.html#123

Reply:
Excellent catch. Thanks.